contracts/HeapOrdering.sol

```diff
@@ -288,24 +263,27 @@ library HeapOrdering {
     /// @param _id The address of the account.
     /// @return The value of the account.
     function getValueOf(HeapArray storage _heap, address _id) internal view returns (uint256) {
-        uint256 rank = _heap.ranks[_id];
-        if (rank == 0) return 0;
-        else return getAccount(_heap, rank).value;
+        uint256 index = _heap.indexes[_id];
+        if (index >= _heap.accounts.length) return 0;
+        Account memory account = _heap.accounts[index];
+        if (account.id != _id) return 0;
+        else return account.value;
     }
 
     /// @notice Returns the address at the head of the `_heap`.
     /// @param _heap The heap to get the head.
     /// @return The address of the head.
     function getHead(HeapArray storage _heap) internal view returns (address) {
-        if (_heap.accounts.length > 0) return getAccount(_heap, ROOT).id;
+        if (_heap.accounts.length > 0) return _heap.accounts[ROOT].id;
         else return address(0);
```

Comment:
at worst, `_heap.accounts[ROOT]` is uninitialized and accessing `_heap.accounts[ROOT].id` reads a null byte which is cast to `address(0)` so we could simplify this as `return _heap.accounts[ROOT].id`